src/components/top-menu/top-menu.ts

```diff
@@ -0,0 +1,195 @@
+import { NameRoute } from '@/enums/name-route';
+import { Container } from '@/components/container/container';
+import { BaseComponent } from '@/templates/base-component';
+import { Link } from '@/templates/link';
+import './top-menu.scss';
+
+const favouritesLink = { href: NameRoute.Favourites };
+const brandsLink = { href: NameRoute.Brands };
+const allLink = { href: NameRoute.Catalog };
+
+export class TopMenu extends BaseComponent {
+  container: Container;
+  private dropDownCategory: BaseComponent;
+  private dropDownSort: BaseComponent;
+  private changeView: BaseComponent;
+  private favouritesLink: Link;
+  private brandsLink: Link;
+  private allLink: Link;
+  private changeColumnsView: BaseComponent;
+
+  constructor(private onclick?: () => void) {
+    super('div', { className: 'catalog__top_menu top_menu' });
+
+    this.container = new Container('top_menu__container');
+    this.append(this.container);
+
+    const topMenuLeft = new BaseComponent('div', { className: 'top_menu__left' });
+    this.container.append(topMenuLeft);
+
+    this.allLink = new Link({ ...allLink, className: 'top_menu__left__item all_link' });
+    topMenuLeft.append(this.allLink);
+
+    const all = new BaseComponent('div', { className: 'top_menu__left__item' });
+    all.setContent('All');
+    this.allLink.append(all);
+
+    this.brandsLink = new Link({ ...brandsLink, className: 'top_menu__left__item brands_link' });
+    topMenuLeft.append(this.brandsLink);
+
+    const brands = new BaseComponent('div', { className: 'top_menu__left__item' });
+    brands.setContent('Brands');
+    this.brandsLink.append(brands);
+
+    const category = new BaseComponent('div', { className: 'top_menu__left__item category' });
+    category.setContent('Category');
+    topMenuLeft.append(category);
+
+    this.dropDownCategory = new BaseComponent('span', { className: 'drop_down' });
+    category.append(this.dropDownCategory);
+
+    const dropDownMenuCategory = new BaseComponent('ul', { className: 'drop_down_menu category' });
+    this.dropDownCategory.append(dropDownMenuCategory);
+
+    const menuItemCleanser = new BaseComponent('li', { className: 'menu_item cleanser' });
```

Comment:
What if we would have more items? Create an array with objects for such cases and those objects would contain classname, content, etc. for each item.